packages/editor-ui/src/views/WorkflowsView.vue

```diff
@@ -224,12 +224,20 @@ export default mixins(showMessage, debounceHelper, newVersions).extend({
 			}
 		},
 		async initialize() {
-			this.usersStore.fetchUsers(); // Can be loaded in the background, used for filtering
-
-			return await Promise.all([
+			await Promise.all([
+				this.usersStore.fetchUsers(),
 				this.workflowsStore.fetchAllWorkflows(),
 				this.workflowsStore.fetchActiveWorkflows(),
 			]);
+
+			// If the user has no workflows and is not participating in the demo experiment,
+			// redirect to the new workflow view
+			if (!this.isDemoTest && this.allWorkflows.length === 0) {
```

Comment:
have you considered whether the user owns the workflow or not? would not `allWorkflows` include workflows that were shared with that user?

Reply:
I don't think there's a difference. Currently, there's no way for a new user, for which this feature is primarily aimed, to own a workflow, so the `allWorkflows` would always be 0 for them. And IMO it would be bad to redirect an owner of instance which 0 workflows to the canvas(if we'd do the logic only on owned workflows) since then they couldn't see ale the workflows on the instance

Reply:
that makes sense 👍🏽 thanks for pointing it out

Reply:
Of course 🙌